### ISSUE TITLE
Dose the learning rate increase or decrease in Adam? 

### ISSUE THREAD

**Issue by AustinApple** [keras-team/keras#12900]
Hello, I use optimizer Adam and try to plot the learning rate in the tensorboard every epoch.
Following is the code I used in the callbacks
~~~~
class SubTensorBoard(TensorBoard):
    def __init__(self, *args, **kwargs):
        super(SubTensorBoard, self).__init__(*args, **kwargs)

    def lr_getter(self):
        # Get vals
        decay = self.model.optimizer.decay
        lr = self.model.optimizer.lr
        iters = self.model.optimizer.iterations # only this should not be const
        beta_1 = self.model.optimizer.beta_1
        beta_2 = self.model.optimizer.beta_2
        # calculate
        lr = lr * (1. / (1. + decay * K.cast(iters, K.dtype(decay))))
        t = K.cast(iters, K.floatx()) + 1
        lr_t = lr * (K.sqrt(1. - K.pow(beta_2, t)) / (1. - K.pow(beta_1, t)))
        return np.float32(K.eval(lr_t))

    def on_epoch_end(self, episode, logs = {}):
        logs.update({"lr": self.lr_getter()})
        super(SubTensorBoard, self).on_epoch_end(episode, logs)
~~~~
the optimizer setting is 
`optimizer = Adam(lr=0.000456, beta_1=0.9902764)`
and learning rate I plot in the following picture
![batch_2048_lr](https://user-images.githubusercontent.com/42490261/58689920-ee7f8100-83ba-11e9-93b0-340f0b664a3c.png)
However, according to the code, lr_t should be updated every epoch like this
`lr_t = lr * (K.sqrt(1. - K.pow(beta_2, t)) / (1. - K.pow(beta_1, t)))`
the picture would be like
<img width="1100" alt="螢幕快照 2019-05-31 下午3 53 21" src="https://user-images.githubusercontent.com/42490261/58690529-584c5a80-83bc-11e9-8d6f-6673ed3b69f2.png">
The value of learning rate is supposed to decay to the learning rate I set initially. But in my case, it start from a point and slowly increase to the learning rate I set.
It really comfuses me.
Does anyone can explain it? Thanks.

**Comment by Zhell1** [comment]
I am currently exploring the same question.

here is what I found which at least explains what you see.
the decreasing rate is between 0 and 1, above 1 (and your t starts at 1), the value increase towards the lr=0.000456

here is a visual explanation of it : 
![adamlr](https://user-images.githubusercontent.com/35771353/58814165-91d3ce80-8625-11e9-995c-257a3e9f3a13.png)

Your two graphs do represent the same function, just at different scale.

I am still wondering too why the lr_t incrases in Adam and how that can allow convergence, but at least it's logical mathematically wise
